src/main/java/MealDemo/rest/MealPlanningRESTController.java

```diff
@@ -0,0 +1,66 @@
+package MealDemo.rest;
+
+import MealDemo.entity.Frequency;
+import MealDemo.entity.Meal;
+import MealDemo.service.MealPlanningServiceImpl;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/test")
+public class MealPlanningRESTController {
+
+    private MealPlanningServiceImpl mealPlanningServiceImpl;
+
+    public MealPlanningRESTController(MealPlanningServiceImpl mealPlanningServiceImpl) {
+        this.mealPlanningServiceImpl = mealPlanningServiceImpl;
+    }
+
+
+
+    @GetMapping("/viewM")
+    public List<Meal> viewAllMeals(){
+        return mealPlanningServiceImpl.getMeals();
+    }
+
+
+    @GetMapping("/viewF")
+    public List<Frequency> viewAllFreq(){
+        return mealPlanningServiceImpl.getFrequencies();
+    }
+
+
+
+
+    @GetMapping("/meals/{mealId}")
+    public Optional<Meal> findByMealById(@PathVariable int mealId){
```

Comment:
Does this work? Can you show me the reponses for:

1. When it finds a meal
2. When it doesn't find a meal.

Include the response codes please. My spring is rusty so I am curious how spring translates the `Optional`

Reply:
The response when it didn't find any meal was  `null` (not great)  
Modified the code to handle _not found_ better, pointing out the problem. 

**Final Results** 
1. When it finds the meal: 
![Screenshot 2023-09-14 at 11 44 21](https://github.com/Natia-bit/MealDemo/assets/82657219/56f32557-6bd0-4c54-a14e-e4ff90a932b6)

2. When it doesn't find a meal: 
![Screenshot 2023-09-14 at 11 45 00](https://github.com/Natia-bit/MealDemo/assets/82657219/d09b956c-be64-42f0-b9b3-31848431f441)

Reply:
That's cool. Didn't know sprint did that for you . . .nice.